awswrangler/distributed/ray/datasources/arrow_orc_datasource.py

```diff
@@ -0,0 +1,62 @@
+"""Ray ArrowCSVDatasource Module."""
+from typing import Any, Dict, List, Optional
+
+import pyarrow as pa
+from pyarrow import orc
+from ray.data.block import BlockAccessor
+
+from awswrangler._arrow import _add_table_partitions, _df_to_table
+from awswrangler.distributed.ray.datasources.pandas_file_based_datasource import PandasFileBasedDatasource
+
+
+class ArrowORCDatasource(PandasFileBasedDatasource):
+    """ORC datasource, for reading and writing ORC files using PyArrow."""
+
+    _FILE_EXTENSION = "orc"
+
+    def _read_file(  # type: ignore[override]
+        self,
+        f: pa.NativeFile,
+        path: str,
+        path_root: str,
+        **reader_args: Any,
+    ) -> pa.Table:
+        columns: Optional[List[str]] = reader_args.get("columns", None)
+
+        table: pa.Table = orc.read_table(f, columns=columns)
+
+        table = _add_table_partitions(
+            table=table,
+            path=f"s3://{path}",
+            path_root=path_root,
+        )
+
+        return table
+
+    def _open_input_source(
```

Comment:
I think this method isn't actually used, we can remove it from the base data source, too.

Reply:
When I remove this I get:
```
(_execute_read_task_split pid=56296) Traceback (most recent call last):
(_execute_read_task_split pid=56296)   File "python/ray/_raylet.pyx", line 641, in ray._raylet.execute_dynamic_generator_and_store_task_outputs
(_execute_read_task_split pid=56296)   File "python/ray/_raylet.pyx", line 2498, in ray._raylet.CoreWorker.store_task_outputs
(_execute_read_task_split pid=56296)   File "/Users/lutleon/Projects/github.com/aws-sdk-pandas/.venv/lib/python3.9/site-packages/ray/data/_internal/lazy_block_list.py", line 685, in _execute_read_task_split
(_execute_read_task_split pid=56296)     for block in blocks:
(_execute_read_task_split pid=56296)   File "/Users/lutleon/Projects/github.com/aws-sdk-pandas/.venv/lib/python3.9/site-packages/ray/data/datasource/datasource.py", line 185, in __call__
(_execute_read_task_split pid=56296)     for block in result:
(_execute_read_task_split pid=56296)   File "/Users/lutleon/Projects/github.com/aws-sdk-pandas/.venv/lib/python3.9/site-packages/ray/data/datasource/file_based_datasource.py", line 466, in read_files
(_execute_read_task_split pid=56296)     for data in read_stream(f, read_path, **reader_args):
(_execute_read_task_split pid=56296)   File "/Users/lutleon/Projects/github.com/aws-sdk-pandas/.venv/lib/python3.9/site-packages/ray/data/datasource/file_based_datasource.py", line 229, in _read_stream
(_execute_read_task_split pid=56296)     yield self._read_file(f, path, **reader_args)
(_execute_read_task_split pid=56296)   File "/Users/lutleon/Projects/github.com/aws-sdk-pandas/awswrangler/distributed/ray/datasources/arrow_orc_datasource.py", line 26, in _read_file
(_execute_read_task_split pid=56296)     table: pa.Table = orc.read_table(f, columns=columns)
(_execute_read_task_split pid=56296)   File "/Users/lutleon/Projects/github.com/aws-sdk-pandas/.venv/lib/python3.9/site-packages/pyarrow/orc.py", line 308, in read_table
(_execute_read_task_split pid=56296)     result = ORCFile(source).read(columns=columns)
(_execute_read_task_split pid=56296)   File "/Users/lutleon/Projects/github.com/aws-sdk-pandas/.venv/lib/python3.9/site-packages/pyarrow/orc.py", line 40, in __init__
(_execute_read_task_split pid=56296)     self.reader.open(source)
(_execute_read_task_split pid=56296)   File "pyarrow/_orc.pyx", line 252, in pyarrow._orc.ORCReader.open
(_execute_read_task_split pid=56296)   File "pyarrow/io.pxi", line 1962, in pyarrow.lib.get_reader
(_execute_read_task_split pid=56296)   File "pyarrow/io.pxi", line 206, in pyarrow.lib.NativeFile.get_random_access_file
(_execute_read_task_split pid=56296)   File "pyarrow/io.pxi", line 235, in pyarrow.lib.NativeFile._assert_seekable
(_execute_read_task_split pid=56296) OSError: only valid on seekable f
```

Reply:
Ok, understood, I suppose it's using base implementation of streaming reads here